spock-spring/src/main/java/org/spockframework/spring/SpringMockTestExecutionListener.java

```diff
@@ -61,6 +64,10 @@ public void beforeTestMethod(TestContext testContext) throws Exception {
     }
   }
 
+  private boolean scanScopedBean(ScanScopedBeans scanScopedBeans, Set<String> scopes, BeanDefinition beanDefinition) {
+    return scanScopedBeans != null && (scopes.size() == 0 || scopes.contains(beanDefinition.getScope()));
```

Comment:
You have already created `scopes` set based on a value of `scanScopedBeans`, it should be enough just to keep `scopes.contains(beanDefinition.getScope())`